Builds fails against musl: libc::ioctl has different argument types

```
unfriend% cargo build --target=x86_64-unknown-linux-musl         
   Compiling reflink v0.1.3
error[E0308]: mismatched types
  --> /home/faux/.cargo/registry/src/github.com-1ecc6299db9ec823/reflink-0.1.3/src/sys/unix.rs:21:39
   |
21 |         libc::ioctl(dest.as_raw_fd(), IOCTL_FICLONE, src.as_raw_fd())
   |                                       ^^^^^^^^^^^^^ expected i32, found u64
```

I happen to be on `rustc 1.36.0-nightly (dec4c5201 2019-05-24)`, Ubuntu x64 host.

--- a/src/sys/unix.rs
+++ b/src/sys/unix.rs
@@ -7,7 +7,7 @@ pub fn reflink(from: &Path, to: &Path) -> io::Result<()> {
     use std::os::unix::io::AsRawFd;
 
     // TODO is this equal on all archs? Just tested on x86_64 and x86.
-    const IOCTL_FICLONE: u64 = 0x40049409;
+    macro_rules! IOCTL_FICLONE { () => (0x40049409) };
 
     let src = fs::File::open(&from)?;
 
@@ -18,7 +18,7 @@ pub fn reflink(from: &Path, to: &Path) -> io::Result<()> {
         .open(&to)?;
     let ret = unsafe {
         // http://man7.org/linux/man-pages/man2/ioctl_ficlonerange.2.html
-        libc::ioctl(dest.as_raw_fd(), IOCTL_FICLONE, src.as_raw_fd())
+        libc::ioctl(dest.as_raw_fd(), IOCTL_FICLONE!(), src.as_raw_fd())
     };
 
     if ret == -1 {

